### PR TITLE
Updated office endpoint to include has-data query parameter

### PR DESCRIPTION
**Pull request by krowvin**
## Issue:  #321 

Added `?has-data=true` query parameter to office endpoint for all offices queries.  

Tested on SWT tomcat instance. Needs to be confirmed against national db. 

Screenshots of this working against SWT
image

image

image

### CODE REVIEW

File: cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java [hunk 1]
```diff
@@ -97,30 +93,19 @@ public void getAll(Context ctx) {
...
-                            @OpenApiContent(from = OfficeFormatV1.class, type = Formats.JSON),
-                            @OpenApiContent(from = TabV1Office.class, type = Formats.TAB),
-                            @OpenApiContent(from = CsvV1Office.class, type = Formats.CSV),
-                            @OpenApiContent(from = CsvV1Office.class, type = Formats.XML)
-                    }
-            )
-            },
-            tags = {"Offices"}
-    )
+    @OpenApi(pathParams = @OpenApiParam(name = "office", description = "The 3 letter office ID you"
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java]: something broke the formatting here. The annotation attributes should each start on a new line for clarity.

File: cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java [hunk 2]
```diff
@@ -53,29 +53,23 @@ public OfficeController(MetricRegistry metrics) {
...
-            },
-            tags = {"Offices"}
-    )
+    @OpenApi(queryParams = {
+            @OpenApiParam(name = "format", deprecated = true, description = "(Deprecated in favor of Accept header) Specifies the encoding "
+                    + "format of the response. Valid value for the format field for this "
+                    + "URI are:\r\n1. tab\r\n2. csv\r\n 3. xml\r\n4. json (default)"),
+            @OpenApiParam(name = "has-data", description = "A flag ('True'/'False') "
+                    + "When set to true this returns offices that have operational data. "
+                    + "Default value is <b>False</b>,. <a href=\"https://github.com/USACE/cwms-radar-api/issues/321\" target=\"_blank\">Feature #321</a>", type = Boolean.class)
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java]: Move type to a new line. Out max line length is 100 (you can run `./gradlew checkstyle` to see other style issues (ignore the files that you didn't touch though.)

Also the original formatting got broken,

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java [hunk 3]
```diff
@@ -6,36 +6,53 @@
 import cwms.radar.data.dto.Office;
 import org.jooq.DSLContext;
 import org.jooq.Record4;
+import org.jooq.Table;
 
 import usace.cwms.db.jooq.codegen.tables.AV_OFFICE;
+import usace.cwms.db.jooq.codegen.tables.AV_LOC2;
 
 public class OfficeDao extends JooqDao<Office>
 {
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java]: Please fix the formatting on these brackets while you're in the file. (Normally I wouldn't worry about it but I'm already asking to to tweak a bunch of other formatting.

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java [hunk 4]
```diff
@@ -6,36 +6,53 @@
...
+
+    public Optional<Office> getOfficeById(String officeId) {
+        AV_OFFICE view = AV_OFFICE.AV_OFFICE;
+        // The .as snippets lets it map directly into the Office ctor fields.
 		Record4<String, String, String, String> fetchOne = dsl.select(view.OFFICE_ID.as("name"), view.LONG_NAME, view.OFFICE_TYPE.as("type"),
 				view.REPORT_TO_OFFICE_ID.as("reportsTo")).from(view).where(view.OFFICE_ID.eq(officeId)).fetchOne();
 		return fetchOne != null ?
 				Optional.of(fetchOne.into(Office.class)) : Optional.empty();
-	}
+    }
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java:83]: Good catch on the indentation.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
Looks good, formatting got a little wonky, probably default settings in your IDE.
The max line length is 100, but always consider if a more vertical layout will be easier to read.

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java [hunk 5]
```diff
@@ -74,7 +74,7 @@ public Optional<Office> getOfficeById(String officeId) {
                 view.OFFICE_TYPE.as("type"),
                 view.REPORT_TO_OFFICE_ID.as("reportsTo"))
             .from(view)
-            .where(view.OFFICE_ID.eq(officeId))
+            .where(view.OFFICE_ID.eq(officeId.toUpperCase()))
```
Comment (krowvin) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java]: Is there a reason we shouldn't be doing this? Confirmed swagger returns an error and claims the office id is not present. Could also set the case in JOOQ _vs_ java toUpperCase

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java (hunk 3 above)
Comment (krowvin) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java]: I went through and fixed all formatting errors that `gradlew checkStyleMain` returned for the two files i've touched, including the javadoc errors.

File: cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java [hunk 6]
```diff
@@ -97,30 +109,28 @@ public void getAll(Context ctx) {
...
-                            @OpenApiContent(from = CsvV1Office.class, type = Formats.CSV),
-                            @OpenApiContent(from = CsvV1Office.class, type = Formats.XML)
-                    }
-            )
-            },
-            tags = {"Offices"}
-    )
+    @OpenApi(pathParams = @OpenApiParam(name = "office", 
+                    description = "The 3 letter office ID you want more information for"), 
+                queryParams = @OpenApiParam(name = "format",
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java]: There's always that one indent. But don't worry about it.

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java (hunk 5 above)
Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java]: Yeah, that should have the JOOQ too upper on both. at the least both *SHOULD* be obviously marked at an upper case comparison.

**Review by MikeNeilson — DISMISSED** [review]
If you want to make the changes for my and your comment go ahead, otherwise this is fine to me.

File: cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java (hunk 6 above)
Comment (krowvin) [reply cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java]: Ahh! I mean i've gone this far!

Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/api/OfficeController.java]: thanks!

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java [hunk 7]
```diff
@@ -74,7 +76,7 @@ public Optional<Office> getOfficeById(String officeId) {
                 view.OFFICE_TYPE.as("type"),
                 view.REPORT_TO_OFFICE_ID.as("reportsTo"))
             .from(view)
-            .where(view.OFFICE_ID.eq(officeId.toUpperCase()))
+            .where(view.OFFICE_ID.eq(upper(officeId)))
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/OfficeDao.java]: should also have view.OFFICE_ID.upper or whatever it actually is. (On the off chance we ever have a non-upper case office ID).

**Review by MikeNeilson — CHANGES_REQUESTED** [review]

File: docs/index.html [hunk 8]
```diff
@@ -0,0 +1 @@
+Hello World
```
Comment (MikeNeilson) [inline docs/index.html]: ?

**Pull request closed without merging**